core/dbt/task/runnable.py

```diff
@@ -206,17 +206,36 @@ def call_runner(self, runner):
         with RUNNING_STATE, uid_context:
             startctx = TimestampNamed('node_started_at')
             index = self.index_offset(runner.node_index)
+            runner.node._event_status['dbt_internal__started_at'] = datetime.utcnow().isoformat()
+            runner.node._event_status['node_status'] = str(RunningStatus.Started)
             extended_metadata = ModelMetadata(runner.node, index)
+
             with startctx, extended_metadata:
-                fire_event(NodeStart(unique_id=runner.node.unique_id))
+                fire_event(
+                    NodeStart(
+                        report_node_data=runner.node,
+                        unique_id=runner.node.unique_id,
+                    )
+                )
             status: Dict[str, str]
             try:
                 result = runner.run_with_hooks(self.manifest)
                 status = runner.get_result_status(result)
+                runner.node._event_status['node_status'] = str(result.status)
+                runner.node._event_status['dbt_internal__finished_at'] = datetime.utcnow().isoformat()
             finally:
-                finishctx = TimestampNamed('node_finished_at')
+                finishctx = TimestampNamed('finished_at')
                 with finishctx, DbtModelState(status):
-                    fire_event(NodeFinished(unique_id=runner.node.unique_id))
+                    fire_event(
+                        NodeFinished(
+                            report_node_data=runner.node,
+                            unique_id=runner.node.unique_id,
+                            run_result=result
+                        )
+                    )
+            del runner.node._event_status["dbt_internal__started_at"]
+            del runner.node._event_status["dbt_internal__finished_at"]
+            del runner.node._event_status["node_status"]
```

Comment:
this is kind of awkward but I wouldn't be surprised if it's necessary. I think it's worth commenting though since it's odd.

Reply:
Jeremy asked for this when we were putting these keys in the config, but we don't need it now that I changed it to put the keys in an event_status dict instead. So they names could be changed back.

Reply:
it's the deleting of all the keys that seems odd to me.

Reply:
I guess I was concerned that there might be situations where the nodes are reused and the status information is leftover from a previous run. I'm not sure if that can happen or not...  Once we're really solid on what information we want to store there we could change it to a class and just reset it.